src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java

```diff
@@ -63,19 +73,26 @@ public void init() throws ServletException {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
-    Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
+    Optional<String> error = validatePostRequest(request);
+    if (error.isPresent()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, error.get());
+      return;
+    }
+
     User author = userService.getCurrentUser();
     if (author == null) {
-      response.sendError(HttpServletResponse.SC_FORBIDDEN, "You are not logged in.");
+      response.sendError(HttpServletResponse.SC_FORBIDDEN, ERROR_NOT_LOGGED_IN);
       return;
     }
+
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
```

Comment:
Order of declaration is very important for readability of a method. I think this can be improved. I would declare in this order (obviously this is not a hard rule).

final Entity commentEntity; // This is the first used variable at line 99
long lectureId ... // used to create lecture which is second var.
Key lecture ... // Second used var
String content ... // Third if you ignore parent and author.
Date dateNow ... // Last
< new line>
Comment.Type type = ... // I declare this right before the if statement so it is obvious to the user where it is used (slightly more important than your other declarations and therefore split from the rest)
if (type == ...)


You can read more about it here: 4.8.2.2 Declared when needed
Local variables are not habitually declared at the start of their containing block or block-like construct. Instead, local variables are declared close to the point they are first used (within reason), to minimize their scope. Local variable declarations typically have initializers, or are initialized immediately after declaration. 

https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations

Reply:
Reordered as you suggested